http-marshallers-java/http-jackson3/src/main/java/org/apache/pekko/http/javadsl/marshallers/jackson3/Jackson.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.pekko.http.scaladsl.model.ErrorInfo;
+import org.apache.pekko.util.ByteString;
+
+import tools.jackson.core.JacksonException;
+import tools.jackson.core.StreamReadConstraints;
+import tools.jackson.core.StreamWriteConstraints;
+import tools.jackson.core.json.JsonFactory;
+import tools.jackson.core.util.BufferRecycler;
+import tools.jackson.core.util.JsonRecyclerPools;
+import tools.jackson.core.util.RecyclerPool;
+import tools.jackson.databind.DeserializationFeature;
+import tools.jackson.databind.ObjectMapper;
+import tools.jackson.databind.json.JsonMapper;
+
+/** A JSON marshaller/unmarshaller using the Jackson library. */
+public class Jackson {
+  private static final ObjectMapper defaultObjectMapper = createMapper();
+
+  /** INTERNAL API */
+  public static class JacksonUnmarshallingException extends ExceptionWithErrorInfo {
+    public JacksonUnmarshallingException(Class<?> expectedType, Exception cause) {
+      super(
+          new ErrorInfo(
+              "Cannot unmarshal JSON as " + expectedType.getSimpleName(), cause.getMessage()),
+          cause);
+    }
+  }
+
+  public static <T> Marshaller<T, RequestEntity> marshaller() {
+    return marshaller(defaultObjectMapper);
+  }
+
+  public static <T> Marshaller<T, RequestEntity> marshaller(ObjectMapper mapper) {
+    return Marshaller.wrapEntity(
+        u -> toJSON(mapper, u), Marshaller.stringToEntity(), MediaTypes.APPLICATION_JSON);
+  }
+
+  public static <T> Unmarshaller<ByteString, T> byteStringUnmarshaller(Class<T> expectedType) {
+    return byteStringUnmarshaller(defaultObjectMapper, expectedType);
+  }
+
+  public static <T> Unmarshaller<HttpEntity, T> unmarshaller(Class<T> expectedType) {
+    return unmarshaller(defaultObjectMapper, expectedType);
+  }
+
+  public static <T> Unmarshaller<HttpEntity, T> unmarshaller(
+      ObjectMapper mapper, Class<T> expectedType) {
+    return Unmarshaller.forMediaType(MediaTypes.APPLICATION_JSON, Unmarshaller.entityToString())
+        .thenApply(s -> fromJSON(mapper, s, expectedType));
+  }
+
+  public static <T> Unmarshaller<ByteString, T> byteStringUnmarshaller(
+      ObjectMapper mapper, Class<T> expectedType) {
+    return Unmarshaller.sync(s -> fromJSON(mapper, s.utf8String(), expectedType));
+  }
+
+  private static String toJSON(ObjectMapper mapper, Object object) {
+    try {
+      return mapper.writeValueAsString(object);
+    } catch (JacksonException e) {
+      throw new IllegalArgumentException("Cannot marshal to JSON: " + object, e);
+    }
+  }
+
+  private static <T> T fromJSON(ObjectMapper mapper, String json, Class<T> expectedType) {
+    try {
+      return mapper.readerFor(expectedType).readValue(json);
+    } catch (JacksonException e) {
+      throw new JacksonUnmarshallingException(expectedType, e);
+    }
+  }
+
+  private static ObjectMapper createMapper() {
+    return createMapper(ConfigFactory.load().getConfig("pekko.http.marshallers.jackson3"));
```

Comment:
is it possible to make it overrideable?

Reply:
The API methods allow you to provide your own ObjectMapper so this code is just for the case where we need to create an ObjectMapper instance because the user hasn't provided one.
If we were to open this up, I would prefer if users were requesting it. This is a copy of the code from pekko-http-jackson and noone was requesting any changes to this for that module.